File: 3751.total-waviness-of-numbers-in-range-i/solution.kt
class Solution {
    fun totalWaviness(num1: Int, num2: Int): Int =
        (maxOf(100, num1) .. num2).sumOf { it.waviness() }
}

fun Int.waviness(): Int =
    toString()
    .windowed(3)
    .count { s -> s[0] < s[1] && s[2] < s[1] || s[0] > s[1] && s[2] > s[1] }

File: 3751.total-waviness-of-numbers-in-range-i/solution.go
func totalWaviness(num1 int, num2 int) int {
    w := 0
    for i := max(100, num1); i <= num2; i++ {
        s := strconv.Itoa(i)
        for j := 1; j < len(s) - 1; j++ {
            if s[j-1] < s[j] && s[j] > s[j+1] || s[j-1] > s[j] && s[j] < s[j+1] {
                w++
            }
        }
    }
    return w
}